js/ui/form/ui.form.layout_manager.js

```diff
@@ -560,6 +570,15 @@ const LayoutManager = Widget.inherit({
         }
 
         const name = item.dataField || item.name;
+        const formOrLayoutManager = this._getFormOrThis();
+
+        const onLabelTemplateRendered = () => {
+            this._incTemplateRenderedCallCount();
+
+            if(formOrLayoutManager.option('templatesRenderAsynchronously') && this._shouldAlignLabelsOnTemplateRendered(itemsCountWithLabelTemplate)) {
```

Comment:
maybe it's metter to move `formOrLayoutManager.option('templatesRenderAsynchronously')` to `_shouldAlignLabelsOnTemplateRendered` to make this code clearer?

Reply:
LGTM